src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -1035,12 +1028,19 @@ p2pTest(const std::shared_ptr<xrt_core::device>& _dev, boost::property_tree::ptr
     const std::string mem_tag(reinterpret_cast<const char *>(mem.m_tag));
     for(const auto& x : sup_list) {
       if(mem_tag.find(x) != std::string::npos && mem.m_used) {
-        if(!p2ptest_bank(_dev->get_device_handle(), _ptTest, mem_tag, static_cast<unsigned int>(midx), mem.m_base_address, mem.m_size << 10, no_dma))
-          break;
-        logger(_ptTest, "Details", mem_tag +  " validated");
+        if(!p2ptest_bank(_dev->get_device_handle(), _ptTest, mem_tag, static_cast<unsigned int>(midx), mem.m_base_address, mem.m_size << 10, no_dma)){
+           run_test.finish(false, "");
+           std::cout << EscapeCodes::cursor().prev_line() << EscapeCodes::cursor().clear_line(); 
+           break;
+	}else {
+          run_test.update(++counter);
+          logger(_ptTest, "Details", mem_tag +  " validated");
+       }
       }
     }
   }
+  run_test.finish(true, "");
```

Comment:
this needs to be wrapped in an if statement. Please remove `run_test.finish(false, "");` L1032-1033 and do the following:
```suggestion
  if(counter < mem_topo->m_count)
    run_test.finish(false, "");
  else
    run_test.finish(true, "");
```